aws/resource_aws_api_gateway_method.go

```diff
@@ -229,6 +240,35 @@ func resourceAwsApiGatewayMethodUpdate(d *schema.ResourceData, meta interface{})
 		})
 	}
 
+	if d.HasChange("authorization_scopes") {
+		o, n := d.GetChange("authorization_scopes")
+		path := "/authorizationScopes"
+
+		old := o.([]interface{})
+		new := n.([]interface{})
+
+		// Remove every authorization scope. Simpler to remove and add new ones,
```

Comment:
Note here: If we switch this to `schema.TypeSet`, there are some set difference functions available: https://godoc.org/github.com/hashicorp/terraform/helper/schema#Set.Difference

It seems like we should try to reduce any downtime associated with updating these if possible.